### PR TITLE
chore: upgrade packages and lint

### PR DESCRIPTION
**Pull request by merlinpaypal**

### CODE REVIEW

File: src/components/modal/v2/parts/Calculator.jsx [hunk 1]
```diff
@@ -157,6 +157,7 @@ const Calculator = ({ setExpandedState, calculator, aprDisclaimer }) => {
                 </div>
             );
         }
+        // eslint-disable-next-line react/jsx-no-useless-fragment
```
Comment (Seavenly) [inline src/components/modal/v2/parts/Calculator.jsx]: Should we consider returning `null` instead?

File: src/components/modal/v2/parts/Calculator.jsx [hunk 2]
```diff
@@ -196,6 +197,7 @@ const Calculator = ({ setExpandedState, calculator, aprDisclaimer }) => {
                     <TermsTable view={view} isLoading={isLoading} aprDisclaimer={aprDisclaimer} />
                 </div>
             ) : (
+                // eslint-disable-next-line react/jsx-no-useless-fragment
```
Comment (Seavenly) [inline src/components/modal/v2/parts/Calculator.jsx]: Should we consider returning `null` instead?

File: src/components/modal/v2/parts/views/LongTerm/Content.jsx [hunk 3]
```diff
@@ -142,6 +140,7 @@ export const LongTerm = ({
                 </Fragment>
             );
         }
+        // eslint-disable-next-line react/jsx-no-useless-fragment
```
Comment (Seavenly) [inline src/components/modal/v2/parts/views/LongTerm/Content.jsx]: Should we consider returning `null` instead?

File: src/components/modal/v2/parts/views/NoInterest/Content.jsx [hunk 4]
```diff
@@ -26,6 +26,7 @@ export const NoInterest = ({
                 </div>
             );
         }
+        // eslint-disable-next-line react/jsx-no-useless-fragment
```
Comment (Seavenly) [inline src/components/modal/v2/parts/views/NoInterest/Content.jsx]: Should we consider returning `null` instead?

File: src/components/modal/v2/parts/views/PayIn1/Content.jsx [hunk 5]
```diff
@@ -25,6 +25,7 @@ export const PayIn1 = ({
                 </Fragment>
             );
         }
+        // eslint-disable-next-line react/jsx-no-useless-fragment
```
Comment (Seavenly) [inline src/components/modal/v2/parts/views/PayIn1/Content.jsx]: Should we consider returning `null` instead?

File: src/components/modal/v2/parts/views/ShortTerm/Content.jsx [hunk 6]
```diff
@@ -50,26 +51,22 @@ export const ShortTerm = ({
...
-                                            currentNum={index + 1}
-                                            timeStamp={installment?.payment_date ?? donutTimestamps[index]}
-                                            numOfPayments={
-                                                (hasInstallments ? estimatedInstallments?.items : donutTimestamps)
-                                                    .length
-                                            }
-                                        />
-                                    )
-                                )}
+                                {}
```
Comment (Seavenly) [inline src/components/modal/v2/parts/views/ShortTerm/Content.jsx]: Is this able to be removed?

File: src/server/message/index.jsx [hunk 7]
```diff
@@ -33,7 +34,7 @@ const applyCascade = curry((style, flattened, type, rules) =>
                 const calculatedVal =
                     typeof val === 'function'
                         ? val({
-                              textSize: style.text?.size * BASIC_FONT_FACTOR
+                              textSize: (style.text?.size || DEFAULT_FONT_SIZE) * BASIC_FONT_FACTOR
```
Comment (Seavenly) [inline src/server/message/index.jsx]: Consider using `??` vs `||` so that only `null` and `undefined` result in the fallback value.

File: src/utils/stats.js [hunk 8]
```diff
@@ -59,8 +59,8 @@ export function buildStatsPayload({ container, activeTags, index, requestDuratio
             // so everything must be converted to a string to prevent unintended filtering
             pos_x: Math.round(containerRect.left).toString(),
             pos_y: Math.round(containerRect.top).toString(),
-            browser_width: (topWindow?.innerWidth).toString(),
-            browser_height: (topWindow?.innerHeight).toString(),
+            browser_width: (topWindow?.innerWidth || '').toString(),
+            browser_height: (topWindow?.innerHeight || '').toString(),
```
Comment (Seavenly) [inline src/utils/stats.js]: We may want to consider using `String(topWindow?.innerWidth)` as a value of "undefined" vs "" will likely be more meaningful.

File: tests/unit/utils/zoidComponentWrapper.js [hunk 9]
```diff
@@ -3,9 +3,13 @@ import { h } from 'preact';
...
-    </XPropsProvider>
-);
+const wrapper = serverData =>
+    // eslint-disable-next-line func-names
+    function ({ children }) {
+        return (
+            <XPropsProvider>
+                <ServerDataProvider data={serverData}>{children}</ServerDataProvider>
+            </XPropsProvider>
+        );
```
Comment (Seavenly) [inline tests/unit/utils/zoidComponentWrapper.js]: Is this change necessary with the added eslint rule `"react/function-component-definition": 0,`?

Comment (merlinpaypal) [reply tests/unit/utils/zoidComponentWrapper.js]: Still gives an error if I remove that line but I've named it `function xpropsProvider({ children })`. Everything else I fixed.

**Review by Seavenly — APPROVED** [review]
